From 1bdb5e2a0561bf1cb52ff0deb02b20b88ef941b1 Mon Sep 17 00:00:00 2001
From: Yotam Ben Dosa <ybendosa@habana.ai>
Date: Sun, 05 Apr 2020 11:02:44 +0300
Subject: [PATCH] [SW-0] openbmc: install-hls: fix part verification

The partition verification function doesn't fully deal with partition
sizes change, as it only tested for cases in which the new partition
image is larger. This is fixed by flashing the full image when the
partition sizes are not equal.

Change-Id: Iaf6d533314ba34d8a869ada6cea08698afe22a2e
Signed-off-by: Yotam Ben Dosa <ybendosa@habana.ai>
---

diff --git a/package/openbmc/install-hls-curr.sh b/package/openbmc/install-hls-curr.sh
index 963bbed..30064f6 100755
--- a/package/openbmc/install-hls-curr.sh
+++ b/package/openbmc/install-hls-curr.sh
@@ -139,7 +139,7 @@
 	local mtd_size=$((16#${mtd_hex_size}))
 	local img_size=$(tar -vtf ${BMC_IMAGE} ${partition_image} | cut -d ' ' -f3)
 
-	if [ ${img_size} -gt ${mtd_size} ]; then
+	if [ ${img_size} -ne ${mtd_size} ]; then
 		echo 0
 	else
 		echo 1
